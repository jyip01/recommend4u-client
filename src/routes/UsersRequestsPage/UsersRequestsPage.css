.newRequestButtonUsers{
  width:50%;
  border:1px solid #384a62;
  color:#384a62;
  display:flex;
  justify-content: center;
  align-items:center;
  padding:5px;
  text-decoration: none;
  font-weight: bold;
  border-radius: 2px;
  margin-top:25px;
}
  
.newRequestButtonUsers:hover{
  background-color: #8bccbc;
  color:white;
  cursor: pointer;
}

.filterBoxUsers{
  display:flex;
  flex-direction:column;
  justify-content: center;
  align-items:center;
}

