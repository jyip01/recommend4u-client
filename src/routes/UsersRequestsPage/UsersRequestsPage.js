import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import RequestApiService from '../../services/request-api-services'
import RequestListContext from '../../contexts/RequestListContext'
import RequestItem from '../../components/RequestItem/RequestItem'
import UserService from '../../services/user-service'
import Error from '../../components/Error/Error'
import './UsersRequestsPage.css'


export default class UsersRequestsPage extends Component {
    
    static contextType = RequestListContext

    // if requestList is empty, make get request for all requests
    // important for refreshing
    componentWillMount=()=>{
        if(!this.context.usersList.length){
            RequestApiService.getRequestsByUserId(UserService.getUserToken())
                .then(this.context.setUsersList)
                .catch(this.context.setError)
        }
    }

    renderRequests=(requests)=>{
        let results
        if(requests.length===1){
            results = '1 Result'
        }
        else{
            results = `${requests.length} Results`
        }
        
        let requestItems = requests.map(request=>
                <RequestItem key={request.id} request={request}/>
            )

        return(
            <>
            <p className="resultsCount">{results}</p>
            <div>{requestItems}</div>
            </>
        )
    }

    componentDidMount=()=>{
        this.context.clearError()
    }
    
    render(){
        return(
            <Error>
                <section className='allRequestsPage'>
                    <div className='filterBox filterBoxUsers'>
                        <h2>YOUR REQUESTS</h2>
                        <Link to='/newrequest' className='newRequestButtonUsers'>New Request</Link> 
                    </div>
                    <div className='requestsBox'>
                        {this.renderRequests(this.context.usersList)}
                    </div> 
                </section>
            </Error>
        )
    }
}
