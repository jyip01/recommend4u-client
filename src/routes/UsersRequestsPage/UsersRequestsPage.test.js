import React from 'react'
import ReactDOM from 'react-dom'
import UsersRequestsPage from './UsersRequestsPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><UsersRequestsPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <UsersRequestsPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
