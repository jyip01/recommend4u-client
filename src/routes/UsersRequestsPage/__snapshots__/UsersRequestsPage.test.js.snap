// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<section
  className="allRequestsPage"
>
  <div
    className="filterBox filterBoxUsers"
  >
    <h2>
      YOUR REQUESTS
    </h2>
    <a
      className="newRequestButtonUsers"
      href="/newrequest"
      onClick={[Function]}
    >
      New Request
    </a>
  </div>
  <div
    className="requestsBox"
  >
    <p
      className="resultsCount"
    >
      0 Results
    </p>
    <div />
  </div>
</section>
`;
